plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1515,7 +1515,13 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
                                 return null;
                             }
 
-                            metastoreClient.revokePrivileges(buildPrivilegeBag(databaseName, tableName, grantee, privilegesToRevoke));
+                            metastoreClient.revokePrivileges(
+                                    buildPrivilegeBag(databaseName, tableName, grantee, privilegesToRevoke),
+                                    privilegesToRevoke.stream()
+                                            .map(PrivilegeGrantInfo::isGrantOption)
+                                            .distinct()
+                                            .findFirst()
```

Comment:
collect to optional? What if there are two options here?

Reply:
Moved to a different implementation.